### PR TITLE
Use a freelist to store render task cache entries, make cache entry public.

### PR DESCRIPTION
**Pull request by glennw**
This is a small patch in preparation for upcoming changes that are
related to caching render tasks. Storing these as a freelist allows
handing out weak handles, rather than always fetching by hash key.

We can provide these to the caller of request_render_task() to allow
fetching the CacheItem during batching, without another hash map
lookup. This is more in line with how the PathFinder integration
works, and also useful for some upcoming patches which will cache
other render task kinds. Specifically, in addition to the CacheItem
for a cached render task, the batching code will be able to retrieve
other details about a cached render task, such as required blend mode.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/2641)

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -798,7 +798,7 @@ dependencies = [
 [[package]]
 name = "osmesa-src"
 version = "17.4.0-devel"
-source = "git+https://github.com/servo/osmesa-src#1cf8c2f21c7e9047c3bd5a2f1accbc08e5636cf7"
+source = "git+https://github.com/jrmuizel/osmesa-src?branch=serialize#0f81d082b4cca16ea4d344bf6758b5a2f8d98277"
```
Comment (kvark) [inline Cargo.lock]: why are we changing the osmesa path here?

Comment (glennw) [reply Cargo.lock]: It's from https://github.com/servo/webrender/pull/2639 - some CI weirdness that @jrmuizel needed temporarily.

**Comment by glennw** [comment]
Yes, handing out weak handles is a follow up to this :)

**Comment by kvark** [comment]
r=me after the CI is green:
> error(B: type alias is never used: `RenderTaskCacheEntryHandle`(B

**Comment by glennw** [comment]
Fixed the warning, thanks.

@bors-servo r=kvark

**Pull request merged**